src/main/java/com/monumental/listeners/MonumentListener.java

```diff
@@ -0,0 +1,52 @@
+package com.monumental.listeners;
+
+import com.monumental.models.Monument;
+import com.monumental.services.MonumentService;
+import org.hibernate.event.spi.PreInsertEvent;
+import org.hibernate.event.spi.PreInsertEventListener;
+import org.hibernate.event.spi.PreUpdateEvent;
+import org.hibernate.event.spi.PreUpdateEventListener;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MonumentListener implements PreInsertEventListener, PreUpdateEventListener {
+
+    @Autowired
+    MonumentService monumentService;
+
+    /**
+     * Before a Monument is inserted, generate its slug
+     */
+    @Override
+    public boolean onPreInsert(PreInsertEvent event) {
+        // This event is called for all inserts so filter out non-Monuments
+        if (!(event.getEntity() instanceof Monument)) {
```

Comment:
I will try to find a way to make this class only run on insert/update of `Monument`s instead of just checking like this